src/main/java/net/greatstart/dao/hibernate/UserDaoImpl.java

```diff
@@ -33,6 +39,23 @@ public User getById(long id) {
 
     @Override
     public List<User> getAll() {
-        return null;
+        List<User> result;
+        SessionFactory sessionFactory;
+        try {
+            sessionFactory = new Configuration().configure().buildSessionFactory();
+        } catch (HibernateException ex) {
+            throw new RuntimeException("Cannot create Session Factory", ex);

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)